pages/en/lb4/Schemas.md

```diff
@@ -9,9 +9,125 @@ summary:
+decorators for adding metadata to your TypeScript/JavaScript
+classes in order to use them with the legacy implementation of the Juggler.
+
+## Using Legacy Juggler
+Defining a schema for use with the legacy juggler involves decorating your
+classes with the `@model` and `@property` decorators.
+
+```ts
+import { model, property } from '@loopback/repository';
+
+@model()
+export class Product extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+    description: 'The unique identifier for a product',
+  })
+  id: number;
+
+  @property({type: 'string'})
+  name: string;
+
+  @property({type: 'string'})
+  slug: string;
+
+  constructor(data?: Partial<Product>) {
+    super(data);
+  }
+}
+```
+
+### Model Decorator
+The model decorator can be used without any additional parameters, or can take
+in a ModelDefinitionSyntax object which follows the general format provided in
+LoopBack 3:
+```ts
+{
+@model({
+  name: "Foo",
+  properties: {
+    // define properties here.
+  },
+  settings: {
+    // etc...
+  }
+})
+class Foo {
+  // etc...
+}
+}
+```
+
+### Property Decorators
+The property decorator takes in the same arguments used in LoopBack 3 for
+individual property entries:
+```ts
+@model()
+class Foo {
+  @property({
+    name: "title",
+    description: "The title of the Foo",
+    type: "string",
+  })
+  public title: string;
+}
+```
+
+The complete list of attributes in property definition can be found in here:
+https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#properties
+
+### Defining Metadata By Hand?
+You might be wondering why we have to redefine the type information when
+it's already a part of the class definition.
+
+The property type data isn't available at runtime, only at compile time.
+This means that LoopBack Next would have to handle this metadata and retrieve
+it at runtime to define your types for you.
+
+While this could be written to work with primitive types, more complex objects
+would require a vastly more involved form of compile-time parsing to build all
+of the metadata required to reconstruct your complex object relationships at
+runtime.
+
+Additionally, even the primitive type handling wouldn't be able to
+determine connector-specific properties you might want to apply to your
+individual types.
+
+```ts
+export class Foo extends Entity {
+  @property name: string; // This would be easy.
+  @property bars: Bar[]; // This would be a lot more difficult!
+  constructor(data?: Partial<Foo>) {
+    super(data);
+  }
+}
+```
+
+## Other ORMs
+You might decide to use an alternative ORM/ODM in your LoopBack application.
+Loopback v4 no longer expects you to provide your data in its own custom Model
+format for routing purposes, which means you are free to alter your classes
+to suit these ORMs/ODMs.
+
+However, this also means that the provided schema decorators will serve no
+purpose for these ORMs/ODMs, and may even conflict with some of them.
```

Comment:
The decorators will attach metadata with the target class under `loopback` namespaces. I don't see much possibility of conflicts.

Reply:
What I meant by this was that other ORMs may have their own `@model` and `@property` decorators.